obspy/io/ascii/core.py

```diff
@@ -470,6 +470,24 @@ def _write_tspair(stream, filename, custom_fmt=None,
                 fh.write(line.encode('ascii', 'strict'))
 
 
+def _determine_dtype(custom_fmt):
+    """
+    :type custom_fmt: str
+    :param custom_fmt: Python string formatter.
+    :rtype: str
+    :return: Datatype string for writing in header. Currently supported
+        are 'INTEGER', 'FLOAT' and `CUSTOM`.
+    """
+    floats = ('e', 'f', 'g')
+    ints = ('d', 'i')
+    if custom_fmt[-1].lower() in floats:
```

Comment:
will raise exception for ```costum_fmt = ''``` - maybe have an ```try/except``` around it and raise a ```ValueError``` if it fails?

Reply:
Or you can replace it with 
```
if custom_fmt[-1:].lower() in floats:
```

in any case an empty format string will fail at some point anyway..

Reply:
Adding `try/except` clause gives a change to provide some more information to user.
But you are right, most of formatters that are left will raise an exception on file writing because of mismatch of np.dtype and string formatter.